CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+you can open a less-formal issue and the maintainers will help you
+turn it into an ADR. 
+
+TODO: ADR registration (eg. in an ADR registration issue)
+
+When the problem as well as proposed solution are well understood,
+changes should start with a [draft
+pull request](https://github.blog/2019-02-14-introducing-draft-pull-requests/)
+against master. The draft signals that work is underway. When the work
+is ready for feedback, hitting "Ready for Review" will signal to the
+maintainers to take a look.
+
+Implementation trajectories should aim to proceed where possible as a series
+of smaller incremental changes, in the form of small PRs that can be merged
+quickly. This helps manage the load for reviewers and reduces the likelihood
+that PRs will sit open for longer.
+
+![Contributing flow](./docs/imgs/contributing.png)
+
+Each stage of the process is aimed at creating feedback cycles which align contributors and maintainers to make sure:
+
+- Contributors don’t waste their time implementing/proposing features which won’t land in `master`.
+- Maintainers have the necessary context in order to support and review contributions.
+
+## Forking
+
+If you do not have write access to the repository, your contribution should be 
+made through a fork on Github. Fork the repository, contribute to your fork,
+and make a pull request back upstream.
+
+When forking, add your fork's URL as a new git remote in your local copy of the
+repo. For instance, to create a fork and work on a branch of it:
+
+- Create the fork on GitHub, using the fork button.
+- `cd` to the original clone of the repo on your machine
+- `git remote rename origin upstream`
+- `git remote add origin git@github.com:<location of fork>
+
+Now `origin` refers to your fork and `upstream` refers to this version.
+Now `git push -u origin master` to update the fork, and make pull requests against this repo.
+
+To pull in updates from the origin repo, run
+
+- `git fetch upstream`
+- `git rebase upstream/master` (or whatever branch you want)
+
+## Changelog
+
+Every non-trivial PR must update the CHANGELOG.md.
+
+Changes for a given release should be split between the four sections: Breaking
+Changes, Features, Improvements, Bug Fixes.
+
+See the [changelog template](CHANGELOG.md). 
+
+Changelog entries should be formatted as follows:
+```
+- [module] \#xxx Some description about the change (@contributor)
+```
+Here, `module` is the part of the code that changed (typically a
+top-level package), `xxx` is the pull-request number, and `contributor`
+is the author/s of the change.
+
+It's also acceptable for `xxx` to refer to the relevent issue number, but pull-request
+numbers are preferred.
+Note this means pull-requests should be opened first so the changelog can then
+be updated with the pull-request's number.
+There is no need to include the full link, as this will be added
+automatically during release. But please include the backslash and pound, eg. `\#2313`.
```

Comment:
> as this will be added automatically during release

That is not the case (yet). In tendermint go there is a python script which does this. Not here.